servicetalk-loadbalancer-experimental/src/main/java/io/servicetalk/loadbalancer/DefaultHostPriorityStrategy.java

```diff
@@ -87,43 +87,28 @@ private <T extends PrioritizedHost> List<T> rebuildWithPriorities(final List<T>
             return hosts;
         }
 
-        // We require that we have a continuous priority set. We could relax this if we wanted by using a tree map to
-        // traverse in order. However, I think it's also a requirement of other xDS compatible implementations.
         List<T> weightedResults = new ArrayList<>();
         int remainingProbability = 100;
-        for (int i = 0; i < groups.size() && remainingProbability > 0; i++) {
-            Group group = groups.get(i);
-            if (group.hosts.isEmpty()) {
-                // we don't have a continuous priority group. Warn and rebuild without priorities.
-                LOGGER.warn("Non-continuous priority groups: {} total groups but missing group {}. " +
-                        "Dropping priority grouping data.", groups.size(), i);
-                return hosts;
-            }
-
-            // We need to compute the weight for the group.
+        for (Group group : groups.values()) {
+            assert !group.hosts.isEmpty();
             final int groupProbability = Math.min(remainingProbability,
                     group.healthPercentage * 100 / totalHealthPercentage);
-            if (groupProbability == 0) {
-                // TODO: this means all hosts for this group are unhealthy. This may be worth some logging.
-            } else {
+            if (groupProbability > 0) {
                 remainingProbability -= groupProbability;
                 group.addToResults(groupProbability, weightedResults);
             }
+            if (remainingProbability == 0) {
+                break;
+            }
         }
-        // What to do if we don't have any healthy nodes at all?
-        if (remainingProbability > 0) {
-            // TODO: this is an awkward situation. Should we panic and just discard priority information?
+        if (weightedResults.isEmpty()) {
```

Comment:
we should have ways to observe these for our observability story.

Reply:
Good point. I'll add a log message for now. If you meant something like an observer then I'm not sure how that would fit nicely but we can look at it as a follow up.